Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1295,17 +1295,29 @@ void LocalFrame::didAccessWindowProxyPropertyViaOpener(WindowProxyProperty prope
 {
     if (m_accessedWindowProxyPropertiesViaOpener.contains(property))
         return;
-    m_accessedWindowProxyPropertiesViaOpener.add(property);
 
-    RefPtr parentWindow { opener() ? opener()->window() : nullptr };
-    if (!parentWindow)
+    auto origin = SecurityOriginData::fromFrame(this);
+    if (origin.isNull() || origin.isOpaque())
+        return;
+
+    if (!opener())
+        return;
+
+    auto openerFrame = dynamicDowncast<LocalFrame>(opener()->mainFrame());
+    if (!openerFrame)
+        return;
+
+    auto parentTopOrigin = SecurityOriginData::fromFrame(openerFrame);
```

Comment:
Where does the term `parent` come in here? Aren't we just getting the origin of `openerMainFrame`? If so, I think this should be called `openerMainFrameOrigin`.